ehrql/query_engines/trino.py

```diff
@@ -138,9 +138,18 @@ def create_inline_table(self, columns, rows):
 
     def reify_query(self, query):
         table_name = f"ehrql_{self.global_unique_id}_tmp_{self.get_next_id()}"
+        # Doesn't work with compound select queries
+        # query = query.add_columns(self.backend.HASHED_ORG_COLUMN)
```

Comment:
You can work around this by wrapping the query up as a subquery:
```py
query = sqlalchemy.select(*query.alias().columns)
```

I just tried this locally, and added the line:
```
query.add_columns(sqlalchemy.literal("fun_times").label("with_access_guard"))
```
And it passed the spec tests, which doesn't mean it _definitely_ works but is a good start.